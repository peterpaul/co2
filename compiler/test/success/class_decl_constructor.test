

["stdio.h"] int printf(char * format, ...);

class MyObject
{
	MyObject(int value) {
		var = value;
	}

	MyObject.copy(MyObject other) {
		var = other.var;
	}

	int var;
	int fun(int a, int b)
	{
		int c = a + b + var;
		return c;
	}
}

int main(int argc, char[][] argv)
{
	MyObject test = new MyObject(5);
	int val = test.var;
	int result = test.fun(val,5);
	printf("result (15) = %d\n", result);
	MyObject testje = new MyObject.copy(test);
	printf("testje.var (5) = %d\n", testje.var);
	return 0;
}