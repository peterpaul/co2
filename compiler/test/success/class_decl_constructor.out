result (15) = 15
testje.var (5) = 5
